liberty-maven-plugin/src/it/dev-it/src/test/java/net/wasdev/wlp/test/dev/it/BaseDevTest.java

```diff
@@ -0,0 +1,215 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package net.wasdev.wlp.test.dev.it;
+
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Properties;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.maven.shared.utils.io.FileUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class BaseDevTest {
+
+   File tempProj;
+   File basicDevProj;
+   File logFile;
+   boolean isWindows = false;
+
+   @Test
+   public void testing() throws Exception {
+      assertTrue(true);
+   }
+  
+   @Before
+   public void setUp() throws Exception {
+      String os = System.getProperty("os.name");
+      if (os != null && os.toLowerCase().startsWith("windows")) {
+         isWindows = true;
+      }
+      
+      if (!isWindows) { // skip tests on windows until server.env bug is fixed
+         
+         tempProj = Files.createTempDirectory("temp").toFile();
+         assertTrue(tempProj.exists());
+         
+         basicDevProj = new File("../resources/basic-dev-project");
+         assertTrue(basicDevProj.exists());
+
+         FileUtils.copyDirectoryStructure(basicDevProj, tempProj);
+         assertTrue(tempProj.listFiles().length > 0);
+         
+         logFile = new File(basicDevProj, "/logFile.txt");
+         logFile.createNewFile();
+         
+         String pluginVersion = System.getProperty("testing");
+
+         File pomXML = new File(tempProj, "/pom.xml");
+         assertTrue(pomXML.exists());
+         
+         Path path = pomXML.toPath();
+         Charset charset = StandardCharsets.UTF_8;
+
+         String content = new String(Files.readAllBytes(path), charset);
+         content = content.replaceAll("SUB_VERION", pluginVersion);
+         Files.write(path, content.getBytes(charset));
+      }
+   }
+
+   @After
+   public void cleanUp() throws Exception {
+      if (!isWindows) { // skip tests on windows until server.env bug is fixed
+         
+         ProcessBuilder builder = new ProcessBuilder();
+         builder.directory(tempProj);
+         String processCommand = "mvn liberty:stop-server";
+
+         String os = System.getProperty("os.name");
```

Comment:
This section of code is used a few times throughout this file.  Might help to move it into a method.